tests/common/test_utils.py

```diff
@@ -70,20 +70,6 @@ async def test_wait_tasks_empty():
     assert len(results) == 0
 
 
-@pytest.mark.asyncio
-async def test_wait_tasks_ok(event_loop, simple_task_list):
-    results = await utils.wait_get_results(*simple_task_list)
-
-    assert len(results) == 1
-    assert results.pop() == (12, 42)
-
-
-@pytest.mark.asyncio
-async def test_wait_tasks_exception(event_loop, task_list_with_exception):
-    with pytest.raises(ValueError):
-        await utils.wait_get_results(*task_list_with_exception)
```

Comment:
@dalf I'm unsure, can we remove this functions?

Reply:
I'm not sure why you want to remove these tests.
Either the issue should is fixed or the current dependencies are kept as they are.

Guess without test: remove async from task_list_with_exception and simple_task_list fixtures.

Reply:
> Either the issue should is fixed or the current dependencies are kept as they are.

My intention was to run CI tests in py3.11 .. but this needs to update some dependencies in `requirements-dev.txt`.

The issues with this functions are:

```
==================================================== FAILURES ====================================================
_______________________________________________ test_wait_tasks_ok _______________________________________________

event_loop = <_UnixSelectorEventLoop running=False closed=False debug=False>
simple_task_list = <coroutine object simple_task_list at 0x7fec71a34940>

    @pytest.mark.asyncio
    async def test_wait_tasks_ok(event_loop, simple_task_list):
>       results = await utils.wait_get_results(*simple_task_list)
E       TypeError: wait_get_results() argument after * must be an iterable, not coroutine

tests/common/test_utils.py:75: TypeError
___________________________________________ test_wait_tasks_exception ____________________________________________

event_loop = <_UnixSelectorEventLoop running=False closed=False debug=False>
task_list_with_exception = <coroutine object task_list_with_exception at 0x7fec72ab8940>

    @pytest.mark.asyncio
    async def test_wait_tasks_exception(event_loop, task_list_with_exception):
        with pytest.raises(ValueError):
>           await utils.wait_get_results(*task_list_with_exception)
E           TypeError: wait_get_results() argument after * must be an iterable, not coroutine

tests/common/test_utils.py:84: TypeError
```

> Guess without test: remove async from task_list_with_exception and simple_task_list fixtures.

No, this won't work since there is an ``await`` in function's block.

> I'm not sure why you want to remove these tests.

TBH .. IDK how to fix:

      TypeError: wait_get_results() argument after * must be an iterable, not coroutine

for the arguments `simple_task_list` ...

https://github.com/searxng/searx-space/blob/c1fa273dc69fa298848eaefee24f26f19c3cbe1f/tests/common/test_utils.py#L45-L49

and `task_list_with_exception` ..

https://github.com/searxng/searx-space/blob/c1fa273dc69fa298848eaefee24f26f19c3cbe1f/tests/common/test_utils.py#L37-L42

Reply:
I just tried, it works: 
* keep the tests
* remove `async` from the fixtures: we can remove async from the function definition, since the bodies of the function are not async.

```
@pytest.fixture
def task_list_with_exception(event_loop):
    tasks = []
    tasks.append(utils.create_task(event_loop, None, dummy_coroutine, 12, second=42))
    tasks.append(utils.create_task(event_loop, None, dummy_raise_exception))
    return tasks


@pytest.fixture
def simple_task_list(event_loop):
    tasks = []
    tasks.append(utils.create_task(event_loop, None, dummy_coroutine, 12, second=42))
    return tasks
```

How I understand: the new version of pytest and pytest-asyncio make a sync call to `simple_task_list` and `task_list_with_exception`. Since the functions are declared async, Python returns a coroutine which raises an error later in `searxstats.common.utils.wait_get_results`.

Reply:
Sorry, I was not paying the needed attention when I read your first comment ... I mixed test functions and the argument functions :facepalm:  .. thanks a lot :+1: